test/modules/TestResultCoordinator/ITestResultComparer.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    /// <summary>
+    /// It is used to compare 2 test results whether they matches or not.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    interface ITestResultComparer<T>
```

Comment:
>ITestResultComparer [](start = 14, length = 19)

Why not use IComparer<T> ?

Reply:
IComparer.Compare method return int to identify whether v1 is greater than or equal to, or less than.  For test result, we just need to know whether it matches or not.  I also don't use IEqualityComparer as it has 2 methods, Equals and GetHashCode; which GetHashCode is not needed for our case.

Reply:
yes, but the int 0 means they are equal. IEqualityComparer can also be helpful becuase first it can check that the hashcodes is equal and then that they are equal. If hashcodes are different there is no need to check equals

---
In reply to: [356759050](https://github.com/Azure/iotedge/pull/2103#discussion_r356759050) [](ancestors = 356759050)